contrib/mesos/pkg/executor/executor.go

```diff
@@ -212,10 +199,15 @@ func (k *KubernetesExecutor) Registered(driver bindings.ExecutorDriver,
 	}
 
 	if executorInfo != nil && executorInfo.Data != nil {
-		k.staticPodsConfig = executorInfo.Data
+		k.initializeStaticPodsSource(executorInfo.Data)
 	}
 
-	k.initialRegistration.Do(k.onInitialRegistration)
+	k.updateChan <- kubelet.PodUpdate{
```

Comment:
possible (highly unlikely) data race between this chan write and the close of the chan in doShutdown (e.g. someone configures a very, very small suicide timeout on a heavily loaded system). solution would be to lock&defer-unlock the state mutex before writing to the chan here.